jenkins-agent.ps1

```diff
@@ -108,11 +108,33 @@ if(![System.String]::IsNullOrWhiteSpace($Cmd)) {
 		$Secret = $env:JENKINS_SECRET
 	}
 	$Secret = $Secret.Trim()
-	if(![System.String]::IsNullOrWhiteSpace($Secret)) {
-		$Secret = " $Secret"
-	}
+
+	$ArgumentList = @(
+		$jnlpProtocolOpts,
+		"-cp",
+		"C:/ProgramData/Jenkins/agent.jar",
+		"hudson.remoting.jnlp.Main",
+		"-headless",
+		$Tunnel,
+		$Url,
+		$WorkDir,
+		$Secret,
+		$Name
+	)
 
 	#TODO: Handle the case when the command-line and Environment variable contain different values.
-	#It is fine it blows up for now since it should lead to an error anyway.
-	Start-Process -FilePath $JAVA_BIN -Wait -NoNewWindow -ArgumentList $("$jnlpProtocolOpts -cp C:/ProgramData/Jenkins/agent.jar hudson.remoting.jnlp.Main -headless$Tunnel$Url$WorkDir$Secret $Name")
+
+	Write-Verbose "Starting to execute Jenkins Agent"
+	$process = Start-Process -FilePath $JAVA_BIN -Wait -NoNewWindow -ArgumentList $ArgumentList
```

Comment:
This seems to produce an error as noted in https://github.com/jenkinsci/kubernetes-plugin/pull/626#issuecomment-545222571.

Reply:
@jglick Is you type `$ArgumentList` when running this when it fails, what's in it? I'm wondering if it's extra whitespace or something else.

Reply:
I did not type anything related. Just (apparently) used the image from this PR by accident, and got a cryptic error. Switching to the regular image (no other changes) fixed it. That is all I know.